### DISCUSSION TITLE
Icon (Set from file) overrides Custom labels

### DISCUSSION THREAD

**Discussion by XLykantX** [cgiesche/streamdeck-homeassistant#221] [Help and Support]
I've got a button set up for controlling my AC device. The setup is as follows:
- Normal (Streamdeck) Titel: AC
- Custom labels ticked and {{hvac_action}} is set to line 3

This gives me the status in line 3 and the title in line 4. There appears to be no icon for climate, at least nothing is displayed. If I set an icon from file in order to get an icon in line 1 and 2 (.png file with transparent background) it doesn't show the hvac action anymore, only the title. That this is also happening if I use pre-defined Streamdeck icons.
It appears to me as if the text of this integration is behind the icon of home assistant.

Maybe I'm missing a step here, I'd be thankful if someone could clarify this :)

Without icon set
![streamdeck_no_icon_set](https://github.com/cgiesche/streamdeck-homeassistant/assets/17988006/f1268a4b-91df-43bd-877d-8007b6e815dd)

With icon set
![streamdeck_icon_set](https://github.com/cgiesche/streamdeck-homeassistant/assets/17988006/bb32fce5-96a1-4428-928c-a3fd54010289)

**Comment by cgiesche** [comment]
Hi XLykantX,

this is a known limitation if you use custom images from the stream-deck application. My plugin can not draw any information on top of custom images (only exception: the cutom title).

My plugin automagically uses the icons that are provided by home-assistant for the entity. Some entities (like your hvac) do not have an icon attribute and thus to not show anything.

There is one thing you could try:
You can customize the icon of entities in home-assistant. This may work in the UI or, as shown below, with configuration entries.

Here is an example how i added a custom icon to the `sun` entity:
```
homeassistant:
  customize:
    sun.sun:
      icon: mdi:hvac
```

This looks like this:
![image](https://github.com/cgiesche/streamdeck-homeassistant/assets/5123180/31f36483-8b2a-4895-b6bc-01ac765124d7)

https://www.home-assistant.io/docs/configuration/customizing-devices/
https://pictogrammers.com/library/mdi/ (list of availabe icons. always prefix with `mdi:` in config)

**Comment by XLykantX** [comment]
Ok, I'll try this :)
Thank you for the quick answer.
